### PR TITLE
Now asking for value of missing environment variables

### PR DESCRIPTION
**Pull request by leo**
If your `env` config property contains an array instead of an object, `now` will ask you for the value of these environment variables. This makes the initial command much shorter (no `-e` needed)!

**BEFORE**

```bash
now -e SLACK_API_TOKEN="<token>" \
    -e SLACK_SUBDOMAIN="<team-name>" \
    -e GOOGLE_CAPTCHA_SECRET="<secret>" \
    -e GOOGLE_CAPTCHA_SITEKEY="<sitekey>" \
    now-examples/slackin
```

**AFTER**

```bash
now now-examples/slackin
```

```json
"env": [
  "SLACK_API_TOKEN",
  "SLACK_SUBDOMAIN",
  "GOOGLE_CAPTCHA_SECRET",
  "GOOGLE_CAPTCHA_SITEKEY"
]
```

Leads to (similar):

screen shot 2017-06-07 at 16 13 54

[This](https://github.com/rauchg/slackin/pull/312) is how we do it.

### CODE REVIEW

File: bin/now-deploy.js [hunk 1]
```diff
@@ -221,6 +222,36 @@ const stopDeployment = msg => {
...
+
+  for (const field of list) {
+    questions.push({
+      name: field,
+      message: field
+    })
+  }
+
+  console.log(
+    '> Please enter the values for the following environment variables:'
```
Comment (matheuss) [inline bin/now-deploy.js]: Please use the `info` helper 🙏

File: bin/now-deploy.js [hunk 2]
```diff
@@ -417,6 +448,13 @@ async function sync({ token, config: { currentTeam, user } }) {
 
   // Merge `now.env` from package.json with `-e` arguments.
   const pkgEnv = nowConfig && nowConfig.env
+
+  if (Array.isArray(nowConfig.env)) {
+    nowConfig.env = await envFields(nowConfig.env)
+  }
+
+  console.log(nowConfig.env)
```
Comment (matheuss) [inline bin/now-deploy.js]: This shouldn't be here right 😅

File: bin/now-deploy.js [hunk 3]
```diff
@@ -417,6 +448,13 @@ async function sync({ token, config: { currentTeam, user } }) {
 
   // Merge `now.env` from package.json with `-e` arguments.
   const pkgEnv = nowConfig && nowConfig.env
+
+  if (Array.isArray(nowConfig.env)) {
```
Comment (matheuss) [inline bin/now-deploy.js]: This breaks deployments that don't have a `now` on `package.json` or a `now.json`:
image

**Review by matheuss — CHANGES_REQUESTED** [review]

**Comment by leo** [comment]
Thank you, @matheuss. Applied your wishes! 😊

**Review by matheuss — APPROVED** [review]
✅

**Review by matheuss — CHANGES_REQUESTED** [review]
👎 
image

**Comment by leo** [comment]
@matheuss Done!

**Pull request merged**